hxd/fs/FileEntry.hx

```diff
@@ -1,34 +1,142 @@
 	public var directory(get, never) : String;
+	/**
+		File extension of the file without the dot (i.e. `png`).
+	**/
 	public var extension(get, never) : String;
+	/**
+		The size of the file in bytes.
+	**/
 	public var size(get, never) : Int;
+	/**
+		Whether this entry is a directory.
+
+		Depending on FileSystem used, it may contain directory entries.
+		In that case they can be iterated over, but should not be accessed as files.
+	**/
 	public var isDirectory(get, never) : Bool;
+	/**
+		Whether this entry is loaded and its contents can be accessed. Otherwise `FileEntry.load` have to be called.
+
+		Depending on FileSystem, some files cannot be accessed immediately and have to be loaded first.
+		In that case `isAvailable` is `false` until file is loaded.
+	**/
 	public var isAvailable(get, never) : Bool;
 
-	// first four bytes of the file
+	/**
+		Returns the first four bytes of the file which usually is a file signature.
+	**/
 	public function getSign() : Int return 0;
 
+	/**
+		Returns the contents of the file as Bytes.
+	**/
 	public function getBytes() : haxe.io.Bytes return null;
 
+	/**
+		Returns the contents of the file as a String.
+	**/
 	public function getText() return getBytes().toString();
 
+	/**
+		Prepares file to be read from the start or moves the reading position to the beginning if its already opened.
+
+		This is an alternative way to read file data without storing its entirety in the RAM.
+		Note that depending on FileSystem used, they always load entire file into RAM, such as EmbedFileSystem or BytesFileSystem.
+		However for LocalFileSystem this is a preferred method of reading files when entirety of the file is not required in the RAM.
+
+		Required to be called before `FileEntry.skip`, `FileEntry.readByte` and `FileEntry.read` can be called.
+		`FileEntry.close` should be called after the file reading is complete.
+	**/
 	public function open() { }
+	/**
+		Moves the reading position of the file by `nbytes`.
+
+		File have to be opened first via `FileEntry.open`.
+	**/
 	public function skip( nbytes : Int ) { }
+	/**
+		Reads a single byte from the file.
+
+		File have to be opened first via `FileEntry.open`.
+	**/
 	public function readByte() : Int return 0;
+	/**
+		Reads the bytes from the file into the output Bytes.
+
+		File have to be opened first via `FileEntry.open`.
+
+		@param out The output Bytes instance file data is written into.
+		@param pos The position in `out` starting from which the data is written.
+		@param size The amount of bytes to read.
+	**/
 	public function read( out : haxe.io.Bytes, pos : Int, size : Int ) {}
+	/**
+		Closes the file previously opened file via `FileEntry.open`.
+	**/
 	public function close() {}
 
+	/**
+		Loads the file, preparing it to be accessed. `FileEntry.isAvailable` can be used to determine is file is loaded/ready to be accessed.
+
+		This is an asynchronous operation and delayed for at least one frame. Only exception is `LocalFileSystem` in macro context.

```

Comment:
No documentation of exception here please. Instead say "might be delayed". These classes are a specification, they should not mention any specific implementation.